### DISCUSSION TITLE
[Bug]: Create socket failed in demo_client

### DISCUSSION THREAD

**Discussion by Y7n05h** [alibaba/xquic#165] [General]
### What happened?

An error occurred with demo_client.

https://github.com/alibaba/xquic/blob/109918f980b3e649f0e7bce4d3bd926a6e44e002/demo/demo_client.c#L1274-L1284

The code save IP and port in `net_cfg.server_addr`. But `xqc_demo_cli_create_socket` use `net_cfg.addr` create socket.

https://github.com/alibaba/xquic/blob/109918f980b3e649f0e7bce4d3bd926a6e44e002/demo/demo_client.c#L1919-L1952

https://github.com/alibaba/xquic/blob/109918f980b3e649f0e7bce4d3bd926a6e44e002/demo/demo_client.c#L1883-L1888

### Steps To Reproduce

Compile with boringSSL on Linux platforms.

1. Run `./demo_server -p 9998` 
2. Run `./demo_client  -p 9998 -a 127.0.0.1`

### Relevant log output

```shell
demo_client Output：

option port :9998
option addr :127.0.0.1
create socket failed, errno: 97
xqc_create_socket error
all tasks are finished, will break loop and exit
```

```
perror 97
OS error code  97:  Address family not supported by protocol
```
```

**Comment by ruiqizhou** [comment]
Duplicate with #128 

The demo_client and demo_server is used for Interoperability Test. You can use . /test_client and . /test_server.
For testing the library, see the [Testing docs](https://github.com/alibaba/xquic/blob/main/docs/docs-zh/Testing-zh.md).

**Comment by vEuphemiav** [comment]
Have you solved this problem? How was it resolved?

**Comment by Yanmei-Liu** [comment]
This has been fixed on the latest main by PR #895. The client now correctly initializes AF_INET/AF_INET6 before creating the socket, preventing socket() from being called with AF_UNSPEC.
